File: UI/Web/src/app/blocks/models/snippet.ts
import { IDimension, Dimension } from './dimension';

export interface ISnippet {
    metadata?: string | null
    dimensions?: IDimension | null
}

export class Snippet implements ISnippet {
    constructor(
        public metadata: string | null = null,
        public dimensions: IDimension | null = new Dimension(),
    ) { }
}

File: system/Infrastructure/Infrastructure/Domain/Snippet.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Infrastructure.Domain
{
    public class Snippet
    {
        public Snippet()
        {
            Dimensions = new Dimensions();
        }

        public string Metadata { get; set; }
        public Dimensions Dimensions { get; set; }

    }
}
